applications/CompressiblePotentialFlowApplication/custom_processes/define_2d_wake_process.cpp

```diff
@@ -0,0 +1,299 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Inigo Lopez
+//
+
+// Project includes
+#include "define_2d_wake_process.h"
+
+namespace Kratos {
+
+// Constructor for Define2DWakeProcess Process
+Define2DWakeProcess::Define2DWakeProcess(ModelPart& rBodyModelPart, const double Tolerance)
+    : Process(), mrBodyModelPart(rBodyModelPart), mTolerance(Tolerance)
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    root_model_part.CreateSubModelPart("trailing_edge_sub_model_part");
+}
+
+void Define2DWakeProcess::ExecuteInitialize()
+{
+    // Set the wake direction and normal for further computations
+    SetWakeDirectionAndNormal();
+    // Save the trailing edge for further computations
+    SaveTrailingEdgeNode();
+    // Check which elements are cut and mark them as wake
+    MarkWakeElements();
+    // Mark the elements touching the trailing edge from below as kutta
+    MarkKuttaElements();
+    // Mark the trailing edge element that is further downstream as wake
+    MartkWakeTrailingEdgeElement();
+}
+
+// This function sets the wake direction and normal for further computations
+const void Define2DWakeProcess::SetWakeDirectionAndNormal()
+{
+    // Reading the free_stream_velocity from the properties
+    const auto free_stream_velocity = mrBodyModelPart.GetProcessInfo().GetValue(FREE_STREAM_VELOCITY);
+    KRATOS_ERROR_IF(free_stream_velocity.size() != 3)
+        << "The free stream velocity should be a vector with 3 components!"
+        << std::endl;
+
+    // Computing the norm of the free_stream_velocity vector
+    const double norm = sqrt(inner_prod(free_stream_velocity, free_stream_velocity));
```

Comment:
```suggestion
    const double norm = std::sqrt(inner_prod(free_stream_velocity, free_stream_velocity));
```
Better use the one in the standard library.